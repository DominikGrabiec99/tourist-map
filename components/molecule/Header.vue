<template>
  <div class="w-full bg-gradient-to-r from-sky-400 to-pink-600 pb-px">
    <div class="h-full w-full bg-white">
      <div class="px-4 py-2 drop-shadow-lg">
        <NuxtLink class="flex items-end" to="/">
          <AtomImage class="h-12" src="favicon.png" alt-text="Tourist App" />
          <AtomTitle
            class="pl-1 text-2xl font-extrabold text-transparent bg-clip-text bg-gradient-to-r from-sky-400 to-pink-600"
            tag="h1"
            content="Tourist App"
          />
        </NuxtLink>
      </div>
    </div>
  </div>
</template>

<script setup lang="ts">
</script>