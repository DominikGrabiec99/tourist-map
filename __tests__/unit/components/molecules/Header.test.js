import { describe, it } from 'vitest';
import { shallowMount } from '@vue/test-utils';

import Header from '@/components/molecule/Header.vue';

describe('Header.vue', () => {
  let wrapper;

  beforeAll(() => {
    wrapper = shallowMount(Header, {
      slots: {
        left: '<div>Left</div>',
        middle: '<div>Middle</div>',
        right: '<div>Right</div>',
        bottom: '<div>Bottom</div>',
      },
    });
  });

  it('render component', () => {
    expect(wrapper.element).toMatchSnapshot();
  });

  it('check slots create', () => {
    expect(wrapper.html()).toContain('Left');
    expect(wrapper.html()).toContain('Middle');
    expect(wrapper.html()).toContain('Right');
    expect(wrapper.html()).toContain('Bottom');
  });
});
