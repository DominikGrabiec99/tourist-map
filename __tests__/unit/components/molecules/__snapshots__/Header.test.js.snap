// Vitest Snapshot v1

exports[`Header.vue > render component 1`] = `
<header
  class="w-full bg-gradient-to-r from-sky-400 to-pink-600 pb-px relative"
>
  <div
    class="h-full w-full flex z-20 relative bg-white dark:bg-neutral-950 transition-color"
  >
    <div
      class="w-full container px-4 mx-auto md:px-14 lg:px-20 drop-shadow-lg py-1 flex justify-between"
    >
      
      <div>
        Left
      </div>
      
      
      <div>
        Middle
      </div>
      
      
      <div>
        Right
      </div>
      
    </div>
  </div>
  
  <div>
    Bottom
  </div>
  
</header>
`;
